packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/utils.js

```diff
@@ -13,29 +13,35 @@
 // limitations under the License.
 
 /**
- * Given a range (`min`, `max`), finds the position of a sub-range (`start`,
- * `end`) factoring in a zoom (`viewStart`, `viewEnd`). The result is returned
- * as a `{ start, end }` object with values ranging in [0, 1].
+ * Given a range (`min`, `max`) and factoring in a zoom (`viewStart`, `viewEnd`)
+ * a function is created that will find the position of a sub-range (`start`, `end`).
+ * The calling the generated method will return the result as a `{ start, end }`
+ * object with values ranging in [0, 1].
  *
  * @param  {number} min       The start of the outer range.
  * @param  {number} max       The end of the outer range.
- * @param  {number} start     The start of the sub-range.
- * @param  {number} end       The end of the sub-range.
  * @param  {number} viewStart The start of the zoom, on a range of [0, 1],
  *                            relative to the `min`, `max`.
  * @param  {number} viewEnd   The end of the zoom, on a range of [0, 1],
  *                            relative to the `min`, `max`.
- * @return {Object}           The resultant range.
+ * @returns {(number, number) => Object} Created view bounds function
  */
-export function getViewedBounds({ min, max, start, end, viewStart, viewEnd }) {
+export function createViewedBoundsFunc({ min, max, viewStart, viewEnd }) {
```

Comment:
You could define and export  type for the output of this function, then in `VirtualizedTraceView` and `SpanBarRow` you can say that `this.getViewedBounds` is of type `viewedBoundsFunctionType` instead of declaring. not crucial, but somewhat DRYer and cleaner.

Reply:
done